logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -171,6 +174,10 @@ fun interface SelfTeamIdProvider {
     suspend operator fun invoke(): Either<CoreFailure, TeamId?>
 }
 
+fun interface SelfConversationIdProvider {
+    suspend operator fun invoke(): Either<StorageFailure, ConversationId>
+}
```

Comment:
How many more `fun interface` will we keep adding to the top of `UserSessionScopeCommon` until we are happy?

Can't we just declare in another file?
Also, implement in another file too so we can write tests for it.

From `UserSessinoScopeCommon` we can do this:

```kotlin
private val selfConversationIdProvider: SelfConversationIdProvider = 
    SelfConversationIdProviderImpl(conversationRepository)
```

`SelfConversationIdProviderImpl` is defined in another file, and properly tested.

Reply:
And now that you're here, could you please add a `TODO` to also refactor and extract `SelfTeamIdProvider ` and `CurrentClientIdProvider`?